crates/l2/utils/eth_client/mod.rs

```diff
@@ -405,7 +405,9 @@ impl EthClient {
                 self.get_chain_id().await?.as_u64()
             },
             nonce: self.get_nonce_from_overrides(&overrides).await?,
-            max_priority_fee_per_gas: overrides.priority_gas_price.unwrap_or_default(),
+            max_priority_fee_per_gas: overrides
+                .priority_gas_price
+                .unwrap_or(self.get_gas_price().await?.as_u64()),
```

Comment:
Use the `if let` form here (as `max_fee_per_gas` for example). `unwrap_or` evaluates the default even if it's not necessary and may cause some problems

Reply:
Fixed, thank you!